packages/cadl-ranch-specs/http/payload/xml/main.tsp

```diff
@@ -0,0 +1,195 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+
+@doc("Sends and receives bodies in XML format.")
+@scenarioService("/payload/xml")
+namespace Payload.Xml;
+
+@doc("Contains fields of primitive types.")
+model SimpleModel {
+  name: string;
+  age: int32;
+}
+
+@doc("Contains fields of arrays of primitive types.")
+model ModelWithSimpleArrays {
+  colors: string[];
+  counts: int32[];
+}
+
+@doc("Contains an array of models.")
+model ModelWithArrayOfModel {
+  items: SimpleModel[];
+}
+
+@doc("Contains an optional field.")
+model ModelWithOptionalField {
+  item: string;
+  counts?: int32;
+}
+
+@scenario
+@route("/simpleModel")
+@scenarioDoc("""
+Returns an instance of SimpleModel.
+Expected response body:
+```xml
+<SimpleModel>
+  <Name>foo</Name>
+  <Age>123</Age>
+</SimpleModel>
+```
+""")
+@get
+op getSimpleModel(): {
+  @header("content-type") contentType: "application/xml";
+  @body body: SimpleModel;
+};
+
+@scenario
+@route("/simpleModel")
+@scenarioDoc("""
+Sends an instance of SimpleModel.
+Expected request body:
+```xml
+<SimpleModel>
+  <Name>foo</Name>
+  <Age>123</Age>
+</SimpleModel>
+```
+""")
+@put
+op putSimpleModel(
+  @header("content-type") contentType: "application/xml",
+  @body input: SimpleModel,
+): void;
+
+@scenario
+@route("/modelWithSimpleArrays")
+@scenarioDoc("""
+Returns an instance of ModelWithSimpleArrays.
+Expected response body:
+```xml
+<ModelWithSimpleArrays>
```

Comment:
By default properties are always wrapped so this should be 

```
<ModelWithSimpleArrays>
  <Colors>
     <string>red<string>
     <string>green<string>
    <string>blue<string>
  </Colors>
  <Counts>
     <string>1<string>
     <string>2<string>
  </Counts>
</ModelWithSimpleArrays>

```

Reply:
I recall that we support both in autorest.testserver. However, I don't know if that was for some hypothetical scenario or if it actually exists.

FWIW, my plan was to also add tests with wrapped arrays. If we don't think we need the unwrapped variant, I can convert the existing ones to be wrapped.

Reply:
Ok I see the default is wrapped. I will convert these and then add tests for the unwrapped variants.

Reply:
fixed